src/v/storage/segment_appender.cc

```diff
@@ -630,11 +630,11 @@ ss::future<> segment_appender::flush() {
         // dispatching background head write.
         auto f = w.p.get_future();
         dispatch_background_head_write();
-        return f;
+        co_return co_await std::move(f);
```

Comment:
it's probably fine, but did you consider any changes to scheduling points that might arise from a conversion to coroutines? since these code paths and the call sites of this predate coroutines by a long time the concern would be caller thinking something is running uninterrupted.

Reply:
ahh ok i missed that. thanks. interesting!